src/averageAges.js

```diff
@@ -55,7 +108,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => onlyWithSon
+    ? people.find(mother => mother.name === child.mother) && child.sex === 'm'
+    : people.find(mother => mother.name === child.mother));
+
+  let age = 0;
+
+  children.forEach(element => {
+    people.find(item => {
+      if (item.name === element.mother) {
+        age += element.born - item.born;
+      }
+    });
+  });
```

Comment:
```suggestion
  children.forEach(element => {
    const mother = people.find(item => item.name === element.mother)
    
    age += element.born - mother.born;
  });
```
Внутри find не делай никаких операций